src/plans/stepsize.jl

```diff
@@ -756,6 +757,64 @@ function show(io::IO, a::NonmonotoneLinesearch)
 end
 get_message(a::NonmonotoneLinesearch) = a.message
 
+@doc raw"""
+    PolyakStepsize <: Linesearch
+
+
+Compute a step size
+
+````math
+α_i = \frac{f(p^{(k)}) - f_{\text{best}}+γ_k}{\lVert{ ∂f(p^{(k)})} \rVert^2},
+````
+
+where ``f_{\text{best}}`` is the (up to now) best cost function seen until now,
+and ``γ_i`` is a sequence of numbers that is square summable, but not summable.
+Furthermore ``∂f`` denotes a subgradient of ``f`` at the current iterate ``p^{(k)}``.
+
+The step size is an adaption from the Dynamic step size discussed in Section 3.2 of [Bertsekas:2015](@cite),
+both to the Riemannian case and to approximate the minimum cost value.
+
+# Fields
+
+* `γ`               : a function `i -> ...` representing the sequence.
+* `best_cost_value` : storing the value `f_{\text{best}}`
+
+# Constructor
+
+    PolyakStepsize(;
+        γ = i -> 1/i,
+        initial_cost_estimate=0.0
+    )
+
+initialize the Polyak stepsize to a certain sequence and an initial estimate of ``f_{\text{best}}``
+"""
+mutable struct PolyakStepsize{F,R} <: Linesearch
```

Comment:
Shouldn't it be a `Stepsize` instead of `Linesearch`?

Reply:
ah Yes. Confused them.